### ISSUE TITLE
xAPI Validation error when `result.extensions` is specified

### ISSUE THREAD

**Issue by wilbrdt** [openfun/ralph#254]
## Bug Report

**Expected behavior/code**
When executing command `validate` on a (known-to-be-correct) xAPI statement, Ralph should return 
`Validating 1 events (ignore_errors=0 | fail-on-unknown=0`

**Actual Behavior**
When executing command `validate` on a (known-to-be-correct) xAPI statement containing **result extensions**, Ralph returns a BadFormatException:
```
2023-01-17 14:25:58,440 INFO     ralph.cli Validating xapi events (ignore_errors=False | fail-on-unknown=False)
2023-01-17 14:25:58,650 ERROR    ralph.models.validator Input event is not a valid VideoSeeked event.
Traceback (most recent call last):
  File "/app/src/ralph/models/validator.py", line 30, in validate
    yield self._validate_event(event_str)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/ralph/models/validator.py", line 77, in _validate_event
    return self.get_first_valid_model(event).json()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/ralph/models/validator.py", line 62, in get_first_valid_model
    raise error
  File "/app/src/ralph/models/validator.py", line 58, in get_first_valid_model
    return model(**event)
           ^^^^^^^^^^^^^^
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 4 validation errors for VideoSeeked
result -> extensions -> https://w3id.org/xapi/video/extensions/length
  extra fields not permitted (type=value_error.extra)
result -> extensions -> https://w3id.org/xapi/video/extensions/played-segments
  extra fields not permitted (type=value_error.extra)
result -> extensions -> https://w3id.org/xapi/video/extensions/progress
  extra fields not permitted (type=value_error.extra)
@timestamp
  extra fields not permitted (type=value_error.extra)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/ralph", line 33, in <module>
    sys.exit(load_entry_point('ralph-malph', 'console_scripts', 'ralph')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/bin/ralph", line 25, in importlib_load_entry_point
    return next(matches).load()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/metadata/__init__.py", line 202, in load
    module = import_module(match.group('module'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/src/ralph/__main__.py", line 23, in <module>
    cli.cli()
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/ralph/cli.py", line 347, in validate
    for event in validator.validate(sys.stdin, ignore_errors, fail_on_unknown):
  File "/app/src/ralph/models/validator.py", line 45, in validate
    raise BadFormatException(message) from err
ralph.exceptions.BadFormatException: Input event is not a valid VideoSeeked event.
```
**Steps to Reproduce**
```
curl -sL https://github.com/openfun/potsie/raw/main/fixtures/elasticsearch/lrs.json.gz | \
gunzip | \
head -n 1 | \
bin/ralph validate -f xapi
```

**Environment**
- Ralph version: 3.1.0
- Platform: Linux

**Comment by wilbrdt** [comment]
The "known-to-be-correct" xAPI statement (taken from lrs.json from Marsha) is actually not following the ADL xAPI video profile for Seeked statement, and is extending the result extensions with:
- https://w3id.org/xapi/video/extensions/length
- https://w3id.org/xapi/video/extensions/played-segments
- https://w3id.org/xapi/video/extensions/progress

Should Ralph still return a validation error for extending the `extensions` field?
We should also probably update lrs.json.gz so that it only contains xAPI statements strictly following the ADL profiles.

**Comment by SergioSim** [comment]
Nice catch! This is indeed unexpected.

TL;DR
I think these events are valid too.

In my understanding, reading [the xAPI profiles spec](https://github.com/adlnet/xapi-profiles/blob/master/xapi-profiles-structure.md#statment-templates):

- A Statement is valid if it matches a Statement Template from the profile.
  - A Statement matches a Statement Template when:
    - it has the same verb, object activity type, attachment usage types, and context activity types
    - it follows all [rules](https://github.com/adlnet/xapi-profiles/blob/master/xapi-profiles-structure.md#81-statement-template-rules) in the Statement Template

Looking at the [video seeked template (v1.0.2)](https://github.com/adlnet/xapi-authored-profiles/blob/master/video/v1.0.2/video.jsonld#L462-L483), there are no `"presence": "excluded"` rules, thus, I think, any additional extensions should be allowed.
